File: java/ArrayARotationOfOther.java
public class ArrayARotationOfOther{

public static boolean isTheSame(int[] array1, int[] array2){
         if(array1.length != array2.length)
             return false;

         int  key = array1[0]; int key_i = -1; int j = -1;

         for(int i = 0; i < array2.length - 1; i++){
             if(array2[i] == key){
                 key_i = i;
                 break;
             }

         }

         if(key_i == -1)
             return false;

         for(int i = 0; i < array1.length - 1; i++){
             j = (key_i + i) % array1.length;
             if(array1[i] != array2[j])
                 return false;
         }

         return true;

     }

     public static void main(String[] args) {
         int[] array1 = {1, 2, 3, 4, 5, 6, 7};
         int[] array2 = {4, 5, 6, 7, 1, 2, 3};

         System.out.println(isTheSame(array1, array2));

     }
}

File: scala/ArrayARotationOfOther.scala
import scala.util.control.Breaks._

class ArrayARotationOfOther{

	def isRotationOfTheOther(array1: Array[Int], array2: Array[Int]): Boolean = {
		if (array1.length != array2.length)	false
		
		val item_i = array1(0); var key_i = -1; var j = -1
		
		breakable
		{
			for((item, index) <- array2.view.zipWithIndex)
				if(item == item_i){
					key_i = index
					break;
				}
		}
		
		if(key_i == -1) false
		
		for((item, index) <- array1.view.zipWithIndex){
			//calculate the index to resume from in the second array
			j = (key_i + index) % array1.length
			if(item != array2(j))
				return false 
		}
		true
		
	}

}



object ArrayARotationOfOther extends App{
	val ar = new ArrayARotationOfOther
	
	val array1 = Array(1, 2, 3, 4)
	val array2 = Array(2, 3, 4, 1)

	println(ar.isRotationOfTheOther(array1, array2))
	
}